cron_scripts/lib/ZCS/API.pm

```diff
@@ -0,0 +1,343 @@
+        }
+        elsif ($err) {
+            chomp($err);
+            last unless ( $try < $maxtry );
+            $sec += $try;    # back off a little more on each retry
+            warn("soap_call $opt{req}: sleep($sec): try#$try error: $err\n");
+            sleep($sec);
+        }
+    }
+
+    if ( $err || $resp->fault ) {
+        $self->Error( "error: $opt{req}"
+              . ( $try > 1 ? " (try #$try)" : "" ) . ": "
+              . ( $err || _err($resp) ) );
+        return undef;
+    }
+
+    return $resp;
+}
+
+sub _err {
+    my ($resp) = @_;
+    return ("no response object") unless ($resp);
+    if ( $resp->fault ) {
+        return (
+            (
+                     $resp->faultcode
+                  || $resp->valueof('/Envelope/Body/Fault/Detail/Error/Code')
+            )
+            . " "
+              . (
+                     $resp->faultstring
+                  || $resp->valueof('/Envelope/Body/Fault/Reason/Text')
+              )
+              . (
+                $Debug
+                ? ( "; "
+                      . $resp->fault->{detail}->{Error}->{Code} . ": "
+                      . $resp->fault->{detail}->{Error}->{Trace} )
+                : ""
+              )
+        );
+    }
+    if ( defined( $resp->valueof("//Fault/faultcode") ) ) {
+        return (
+                $resp->valueof("//Fault/faultcode") . " "
+              . $resp->valueof("//Fault/faultstring")
+              . (
+                $Debug
+                ? ( "; "
+                      . $resp->valueof('//Fault/detail/Error/Code') . ": "
+                      . $resp->valueof('//Fault/detail/Error/Trace') )
+                : ""
+              )
+        );
+    }
+}
+
+# expect no issues with token expiring
+sub auth {
+    my $self = shift;
+    unless ( exists( $self->{_zimbra_auth} ) ) {
+
+        # authenticate with zimbra to get AuthToken
+        my $req = "AuthRequest";
+        my $attr = { "xmlns" => "urn:zimbraAdmin" };
+        my $head =
+          SOAP::Header->name("context")->attr( { "xmlns" => "urn:zimbra" } );
+        my $body = [
+            SOAP::Data->name( name => $self->conf->{SOAPUser} )->type("string"),
+            SOAP::Data->name( password => $self->conf->{SOAPPass} )
+              ->type("string")
+        ];
+
+        my $resp = $self->soap_call(
+            req  => $req,
+            attr => $attr,
+            head => $head,
+            body => $body
+        );
+        die $self->Error unless $resp;
+
+        # Convert authToken into value that can be passed to zimbra requests
+        # and cache for all future requests
+        $self->{_zimbra_auth} = SOAP::Header->name(
+            "context" => \SOAP::Header->value(
+                SOAP::Header->name(
+                    "authToken" => $resp->valueof('//authToken')
+                ),
+            )
+        )->attr( { xmlns => "urn:zimbra" } );
+    }
+    return $self->{_zimbra_auth};
+}
+
+sub delegateauth {
```

Comment:
nit: since you're paring back the module, it sure looks like you could remove all the delegate auth stuff as you don't seem to be using it